### PR TITLE
Add spi-based-sharding-spring-boot-jpa-example for spi-based-sharding-algorithm-example

### LINKED ISSUE #15718: [OpenForce 2022]  Add `spi-based-sharding-spring-boot-jpa-example` for `spi-based-sharding-algorithm-example`
Hi community,

This issue is for [OpenForce 2022](https://github.com/apache/shardingsphere/discussions/12683#discussioncomment-2139441)

### Background

[spi-based-sharding-algorithm-example](https://github.com/apache/shardingsphere/tree/master/examples/shardingsphere-jdbc-example/single-feature-example/extension-example/custom-sharding-algortihm-example/spi-based-sharding-algorithm-example) is an example of a custom algorithm that demonstrates how to integrate a user-defined sharding algorithm and configure it using  [SPI (Service Provider Interface)](https://docs.oracle.com/javase/tutorial/sound/SPI-intro.html).

### Aim

Add `spi-based-sharding-spring-boot-jpa-example` for `spi-based-sharding-algorithm-example`

### Basic Qualifications

- Java
- Maven

### Example FYI

- [sharding-spring-boot-jpa-example](https://github.com/apache/shardingsphere/tree/master/examples/shardingsphere-jdbc-example/single-feature-example/sharding-example/sharding-spring-boot-jpa-example)

### PR DESCRIPTION
**Pull request by RakeshPotnuru**
Fixes #15718 

Changes proposed in this pull request:
- spi based sharding example

### CODE REVIEW

File: examples/shardingsphere-jdbc-example/single-feature-example/extension-example/custom-sharding-algortihm-example/spi-based-sharding-algorithm-example/spi-based-sharding-spring-boot-jpa-example/pom.xml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-data-jpa</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-test</artifactId>
+        </dependency>
+    </dependencies>
+</project>
+
```
Comment (yx9o) [inline examples/shardingsphere-jdbc-example/single-feature-example/extension-example/custom-sharding-algortihm-example/spi-based-sharding-algorithm-example/spi-based-sharding-spring-boot-jpa-example/pom.xml:51]: Please remove extra blank lines.

File: examples/shardingsphere-jdbc-example/single-feature-example/extension-example/custom-sharding-algortihm-example/spi-based-sharding-algorithm-example/spi-based-sharding-spring-boot-jpa-example/src/main/java/org/apache/shardingsphere/example/sharding/spring/boot/jpa/SPIBasedShardingSpringBootJpaExample.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import org.springframework.boot.autoconfigure.transaction.jta.JtaAutoConfiguration;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.ComponentScan;
+
+import java.sql.SQLException;
+
+@ComponentScan("org.apache.shardingsphere.example.core.jpa")
+@EntityScan(basePackages = "org.apache.shardingsphere.example.core.jpa.entity")
+@SpringBootApplication(exclude = JtaAutoConfiguration.class)
+public class ShardingSpringBootJpaExample {
```
Comment (yx9o) [inline examples/shardingsphere-jdbc-example/single-feature-example/extension-example/custom-sharding-algortihm-example/spi-based-sharding-algorithm-example/spi-based-sharding-spring-boot-jpa-example/src/main/java/org/apache/shardingsphere/example/sharding/spring/boot/jpa/SPIBasedShardingSpringBootJpaExample.java:34]: The class name must be the same as the file name.

**Comment by RaigorJiang** [comment]
Hi @RakeshPotnuru 
Please fix the failed CI.

**Review by RaigorJiang — APPROVED** [review]

**Comment by RaigorJiang** [comment]
Merged, thank you all! @RakeshPotnuru @yx9o

**Pull request merged**